homeassistant/components/deconz/binary_sensor.py

```diff
@@ -52,9 +51,11 @@
 class DeconzBinarySensorDescriptionMixin:
     """Required values when describing secondary sensor attributes."""
 
-    suffix: str
+    name_suffix: str
+    unique_id_suffix: str
     update_key: str
-    value_fn: Callable[[SensorResources], bool | None]
+    value_attr: str
+    old_unique_id_suffix: str
 
 
 @dataclass
```

Comment:
It's painful to see sooo much code that could be generative from an array of tuples, but I think the typing system isn't smart enough to unroll a loop that generates this type of dataclass.

Reply:
> It's painful to see sooo much code that could be generative from an array of tuples,

We are coming from that... its even more painful 😉